src/AppInstallerCommonCore/Public/winget/Filesystem.h

```diff
@@ -41,6 +41,10 @@ namespace AppInstaller::Filesystem
     // Returns true if replacement happened, false otherwise.
     bool ReplaceCommonPathPrefix(std::filesystem::path& source, const std::filesystem::path& prefix, std::string_view replacement);
 
+    // If the path begins with a common prefix for APPDATA or LOCALAPPDATA, replace it with the anonymized version
+    // Returns true if replacement happened, false otherwise.
+    bool AnonymizeFilePath(std::filesystem::path& source);

```

Comment:
I don't think this fits as a common util method. We may not be covering all cases. The "replace with env var" thing depends on caller intention, some  may want %Programfiles% replaced, some may want %UserProfile% replaced. By doing multiple attempts in one method here, at least some attempts will be unnecessary. For the GetPathTo() result, I think it would better call ReplaceCommonPathPrefix with corresponding "what to replace" directly inside the GetPathTo() implementation.